main/test/ca/uwaterloo/flix/library/Pkger/TestParser.flix

```diff
@@ -0,0 +1,454 @@
+// take                                                                                      //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def take1(): Bool =
+    String.toList("abc") |> char().take(2) == Some(('c' :: Nil, 'a' :: 'b' :: Nil))
+   
+@test
+def take2(): Bool =
+    String.toList("a") |> char().take(2) == None
+   
+@test
+def take3(): Bool =
+    String.toList("a") |> char().take(0) == Some(('a' :: Nil, Nil))
+
+@test
+def take4(): Bool =
+    String.toList("a") |> char().take(-1) == Some(('a' :: Nil, Nil))
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+// seperatedTake                                                                             //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def seperatedTake1(): Bool =
+    String.toList("a.b.c") |> seperatedTake(char(), tag('.'), 2)
+    ==
+    Some(('.' :: 'c' :: Nil, 'a' :: 'b' :: Nil))
+   
+@test
+def seperatedTake2(): Bool =
+    String.toList("a.") |> seperatedTake(char(), tag('.'), 2)
+    ==
+    None
+   
+@test
+def seperatedTake3(): Bool =
+    String.toList(".a") |> seperatedTake(char(), tag('.'), 0)
+    ==
+    Some(('.' :: 'a' :: Nil, Nil))
+   
+@test
+def seperatedTake4(): Bool =
+    String.toList(".a") |> seperatedTake(char(), tag('.'), -1)
+    ==
+    Some(('.' :: 'a' :: Nil, Nil))
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+// tailSeperatedTake                                                                         //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def tailSeperatedTake1(): Bool =
+    String.toList("a.b.c") |> tailSeperatedTake(char(), tag('.'), 2)
+    ==
+    Some(('c' :: Nil, 'a' :: 'b' :: Nil))
+   
+@test
+def tailSeperatedTake2(): Bool =
+    String.toList("a.") |> tailSeperatedTake(char(), tag('.'), 2) == None
+   
+@test
+def tailSeperatedTake3(): Bool =
+    String.toList(".a") |> tailSeperatedTake(char(), tag('.'), 0) == Some(('a' :: Nil, Nil))
+   
+@test
+def tailSeperatedTake4(): Bool =
+    String.toList(".a") |> tailSeperatedTake(char(), tag('.'), -1) == Some(('a' :: Nil, Nil))
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+// takeAny                                                                                   //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def takeAny1(): Bool =
+    Nil |> takeAny(tag('a')) == Some((Nil, Nil))
+
+@test
+def takeAny2(): Bool =
+    ('a' :: Nil) |> takeAny(tag('a')) == Some((Nil, 'a' :: Nil))
+
+@test
+def takeAny3(): Bool = 
+    ('a' :: 'b' :: Nil) |> takeAny(char().filter(Char.isLetter))
+    ==
+    Some((Nil, 'a' :: 'b' :: Nil))
+
+@test
+def takeAny4(): Bool = 
+
```

Comment:
Why extra new lines?

Reply:
The extra line between the definition and the body of the function was a mistake, but the body has been split up in multiple lines to avoid line wrapping. I've arbitrarily chosen the text width to be 100 characters, since this is a little less that what fits on half my screen with my particular setup, making space for line numbers on each line.